### PR TITLE
test(PAIRING API): Add new tests for agent and credentials

### PR DESCRIPTION
**Pull request by OmarBrbutovic**
* [x] I have read [CONTRIBUTING.md](../CONTRIBUTING.md) and [CODE_OF_CONDUCT.md](../CODE_OF_CONDUCT.md)
* [x] I have added to [CHANGELOG.md](../CHANGELOG.md) relevant changes and any other user facing change
* [x] I have added or ran the appropriate tests

#### What this PR does / why we need it:

- Add tests to verify that when the `details` field is an empty string in a credentials status response, it is correctly interpreted as `nil`.
- Add validation for initial introspection versions:
  - One for a negative major version.
  - One for a negative minor version.

##### Does this PR introduce a user-facing change?
* [ ] Yes
* [x] No

### CODE REVIEW

File: apps/astarte_pairing_api/test/astarte_pairing_api/agent/agent_test.exs [hunk 1]
```diff
@@ -125,6 +125,38 @@ defmodule Astarte.Pairing.API.AgentTest do
...
+
+      attrs = Map.put(@valid_attrs, "initial_introspection", initial_introspection)
+      assert {:error, changeset} = Agent.register_device(@test_realm, attrs)
+
+      messages =
+        Keyword.get_values(changeset.errors, :initial_introspection) |> Enum.map(&elem(&1, 0))
+
+      assert messages == [
+               "has negative versions in interface org.astarteplatform.Values"
+             ]
```
Comment (Annopaolo) [inline apps/astarte_pairing_api/test/astarte_pairing_api/agent/agent_test.exs:141]: Messages might change, as they do not depend on us but on Ecto. Imho the following check is enough for both this and the next test:
```elixir
      assert {:error, _changeset} = Agent.register_device(@test_realm, attrs)
```

**Review by Annopaolo — CHANGES_REQUESTED** [review]

**Review by emirh1 — APPROVED** [review]

**Review by Annopaolo — APPROVED** [review]
